packages/ext-taquito/src/tzktReadProvider.ts

```diff
@@ -0,0 +1,195 @@
+import 'cross-fetch/polyfill';
+import {
+  BlockResponse,
+  EntrypointsResponse,
+  MichelsonV1Expression,
+  SaplingDiffResponse, ScriptedContracts,
+} from '@taquito/rpc';
+import {
+  BigMapQuery,
+  BlockIdentifier,
+  SaplingStateQuery,
+  TzReadProvider
+} from '@taquito/taquito';
+import {
+  accountsGetBalanceAtLevel,
+  accountsGetByAddress,
+  accountsGetCounter, bigMapsGetKey, Block, blocksGet,
+  blocksGetByHash,
+  blocksGetByLevel, blocksGetCount,
+  contractsGetEntrypoints, contractsGetRawStorage,
+  contractsGetStorageSchema, headGet, Protocol, protocolsGetByCycle,
+  protocolsGetCurrent
+} from '@tzkt/sdk-api';
+import {BigNumber} from 'bignumber.js';
+
+
+
+export class TzktReadProvider implements TzReadProvider {
+  constructor(private readProvider: TzReadProvider) {}
+
+  private readonly liveBlockFilters = {
+    level: {},
+    sort: {
+      desc: 'level',
+    },
+    limit: 120, select: {fields: ['hash', 'timestamp']}
+  };
+
+
+  private async _getBlockIdentifier(block: BlockIdentifier): Promise<number> {
+    if(typeof block === 'number') {
+      return block
+    }
+
+    if(block === 'head' || !block) {
+      const {data} = await blocksGetCount();
+      return data
+    }
+
+    if(String(block).includes('head~')) {
+      const n = Number(String(block).split('head~')[1])
+      const {data} = await blocksGetCount();
+
+      return  Number(data) - n
+    }
+
+    const {data} = await blocksGetByHash(block);
+    return data?.level || 0
+  }
+
+  async getEntrypoints(contract: string): Promise<EntrypointsResponse> {
+    const {data} = await contractsGetEntrypoints(contract);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return data;
+  }
+
+  async getScript(address: string): Promise<ScriptedContracts> {
+    const {data} = await contractsGetStorageSchema(address);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return data;
+  }
+
+  async getBalance(address: string, block: BlockIdentifier): Promise<BigNumber> {
+    const blockId = await this._getBlockIdentifier(block);
```

Comment:
If block is head you don't need to do two requests, you can just call getBalance